src/jest.setup.js

```diff
@@ -15,14 +15,13 @@ const { getSuite } = require('sauce-testrunner-utils');
 const { logHelper } = require('./utils');
 debug.log = logHelper
 
-const testTimeout = (parseInt(process.env.TEST_TIMEOUT) || DEFAULT_JEST_TIMEOUT)
+const runCfg = getRunnerConfig();
+const suite = getSuite(runCfg, SUITE_NAME);
+const testTimeout = (suite.timeout / 1000000000 || parseInt(process.env.TEST_TIMEOUT) || DEFAULT_JEST_TIMEOUT)
```

Comment:
Hum, I guess that setting will set a timeout of 30min per test, not a 30min overall.

Reply:
Based on documentation https://jestjs.io/docs/jest-object#jestsettimeouttimeout this should work. `Note: If you want to set the timeout for all test files, a good place to do this is in setupFilesAfterEnv` . We have src/jest.setup.js in setupFilesAfterEnv.